### DISCUSSION TITLE
Nextjs deploy failed. Error: Found usage of dynamic routes but not on a new enough version of Next.js

### DISCUSSION THREAD

**Discussion by yellowspaceboots** [vercel/vercel#4815] [Help]
Nextjs Version from package.json is "next": "^9.4.4" and it runs fine locally in dev but cannot deploy through github. I am confused at the error message saying i am not using a new enough version of Next.js when I have the latest stable version. Full disclosure I am using yarn 2 as my package manager. Any insight into how I can get this to deploy would be great! Thanks!

**Comment by yellowspaceboots** [comment]
I updated my next to the canary build to see if that would fix it and I am now getting 

Error: No Next.js version could be detected in your project. Make sure `"next"` is installed in "dependencies" or "devDependencies"

inside the vercel deployment log.

**Comment by thornyweb** [comment]
I am seeing the same issue using Yarn Berry (v2) in a new Next.js project.

**Reply by iam-yan** [reply]
Have you solved it?

**Reply by thornyweb** [reply]
I had an email exchange directly with Vercel support and they don't currently support Yarn Berry.

**Reply by iam-yan** [reply]
I just found the solution: 
- Update next version to: Next Canary
- Use vercel cli to deploy

( git integration still dose not support yarn berry )

**Reply by kachkaev** [reply]
@thornyweb did the Vercel support folks share any timeline info? There is [new GitHub issue on this](https://github.com/vercel/vercel/issues/5280), but no publicly visible movement or milestones. I seem to be unable to [migrate `njt` to yarn 2](https://github.com/kachkaev/njt/pull/29) until this is resolved.

**Reply by thornyweb** [reply]
@kachkaev No there was no timeline given, looking at other comments though from @juuyan there is a CLI work-around for now if you want / can use that.

**Comment by iam-yan** [comment]
Just found the solution:
- Update next version to: Next Canary
- Use vercel cli to deploy
( git integration still dose not support yarn berry )
